pkg/action/action.go

```diff
@@ -109,9 +110,19 @@ type Configuration struct {
 	// HookOutputFunc called with container name and returns and expects writer that will receive the log output.
 	HookOutputFunc func(namespace, pod, container string) io.Writer
 
+	// logger is an slog.Logger pointer to use with the Configuration instance
+	logger atomic.Pointer[slog.Logger]
+
+	// Mutex is an exclusive lock for concurrent access to the action
 	mutex sync.Mutex
 }
 
+func NewConfiguration() *Configuration {
```

Comment:
```suggestion
func NewConfiguration(options...ConfigurationOptions) *Configuration {
```

Options might be better than this `SetLogger()` pattern?

Reply:
ie. 
```
type ConfigurationOption func(c *Configuration)

// Override the default logging handler
// If unspecified, the default logger will be used
func ConfigurationSetLogger(h *slog.Handler) ConfigurationOption {
    return func(c *Configuration) {
       c.logger = slog.New(h)
    }
}
```

```
func NewConfiguration(options...ConfigurationOption) {
    c := &Configuration{}
    for _, o := options {
       o(c)
    }
    
     if o.logger == nil {
        o.logger = slog.Default()
     }

```

Reply:
Done. It allows us to add more options in future